algebras/algebra/src/main/scala/endpoints4s/algebra/Urls.scala

```diff
@@ -108,6 +108,23 @@ trait Urls extends PartialInvariantFunctorSyntax {
       value: QueryStringParam[A]
   ): QueryString[A]
 
+  /** Builds a `QueryString` with one optional parameter, which has a default value.
+    *
+    * Examples:
+    *
+    * {{{
+    *   optQsWithDefault[Int]("page", 1) // optional `page` parameter, with default value 1
+    * }}}
+    *
+    * @param name Parameter’s name
+    * @tparam A Type of the value carried by the parameter
+    * @group operations
+    */
+  def optQsWithDefault[A](name: String, default: A, docs: Documentation = None)(implicit
+      value: QueryStringParam[A]
+  ): QueryString[A] =
```

Comment:
So, the problem with `QueryString[A]` is that clients will have to provide a value, even though it’s optional, and the problem with `QueryString[Option[A]]` is that servers will have to deal with `None` although in practice there will always be a value.

Note that the situation is the same with `optFieldWithDefault`, which returns a `JsonSchema[A]`. This means that clients have to provide a value. I guess this is not a problem because `optFieldWithDefault` is mostly used via the [generic derivation mechanism](https://github.com/endpoints4s/endpoints4s/blob/19f893087c0f8f9060f14e79e3925b4030dba738/json-schema/json-schema-generic/src/main/scala-2/endpoints4s/generic/JsonSchemas.scala#L134), so eventually, the compiler does provide the default value (when you create a case class with a field that has a default value, the compiler supplies that value for you).

I wonder what we should do here. Maybe introduce a type `WithDefault[A]` which would be `Option[A]` on the client interpreters and `A` on the server interpreters?

Reply:
@julienrf I just opened #941 with, what I understand, is your idea to solve this